### PR TITLE
Support fixes for ROCKS meta-store

### PR DESCRIPTION
**Pull request by ggezer**

### CODE REVIEW

**Comment by ggezer** [comment]
Jenkins, test this please

**Comment by ggezer** [comment]
Jenkins, test this please

File: core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java [hunk 1]
```diff
@@ -80,6 +82,14 @@ public RocksBlockStore(String baseDir) {
         .setCreateMissingColumnFamilies(true);
     String dbPath = PathUtils.concatPath(baseDir, BLOCKS_DB_NAME);
     String backupPath = PathUtils.concatPath(baseDir, BLOCKS_DB_NAME + "-backups");
+    // Create block store db path if it does not exist.
+    if (!FileUtils.exists(dbPath)) {
+      try {
+        FileUtils.createDir(dbPath);
+      } catch (IOException ioe) {
```
Comment (apc999) [inline core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java]: nit: unless `e` is taken, just name exception to be `e` in a `catch` clause

Comment (ggezer) [reply core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java]: Hmm. I didn't know about this convention. I might have broken it many times..

File: core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java [hunk 2]
```diff
@@ -86,8 +86,8 @@ public RocksBlockStore(String baseDir) {
     if (!FileUtils.exists(dbPath)) {
       try {
         FileUtils.createDir(dbPath);
-      } catch (IOException ioe) {
-        LOG.warn("Failed to create nonexistent db path at: {}. Error:{}", dbPath, ioe);
+      } catch (IOException e) {
+        LOG.warn("Failed to create nonexistent db path at: {}. Error:{}", dbPath, e);
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java]: do we need `e.toString` here?

Comment (ggezer) [reply core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java]: It's implicit I think, because it's interpreted as `Object`

Comment (calvinjia) [reply core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java]: I was worried about it treating it differently when the last arg is a throwable, but looking at the implementation I think you are right.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**